### PR TITLE
MEED-353: Add Favorite action in the space topbar popover

### PR DESCRIPTION
**Pull request by SaraBoutej**
This change allows to bookmark a space from the space topbar popover, then we add the favorite button icon in the topbar space popover action area.

### CODE REVIEW

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp [hunk 1]
```diff
@@ -44,6 +47,8 @@
     }
     titleClass = "company";
   } else {
+    spaceObject = spaceService.getSpaceById(space.getId());
+    space = spaceObject;
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp]: redundant IMO
```suggestion
```

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp [hunk 2]
```diff
@@ -71,6 +76,7 @@
   });
   <%} %>
   let params = {
+    space: `<%=spaceObject%>`,
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp]: You need here the space REST object representation and not the space Service Layer DTO. BTW, you may need just the 'isFavorite' attribute only

File: webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue [hunk 3]
```diff
@@ -36,29 +40,29 @@ export default {
...
   },
   data: () => ({
-    isFavorite: false,
+    favorite: false,
   }),
   computed: {
-    spaceId() {
-      return this.space?.id;
+    id() {
+      return this.spaceId;
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue]: no need to compute this IMO

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp [hunk 4]
```diff
@@ -71,6 +89,9 @@
   });
   <%} %>
   let params = {
+    id: `<%=spaceId%>`,
+    isFavorite: `<%=isFavorite%>`,
+    isMember: `<%=isMember%>`,
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp:94]: Can non member access to space logo banner? I think it's an unnecessary check at this stage

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue [hunk 5]
```diff
@@ -36,29 +40,26 @@ export default {
...
   }),
-  computed: {
-    spaceId() {
-      return this.space?.id;
-    },
-  },
   created() {
-    this.isFavorite = this.space?.isFavorite === 'true';
+    this.favorite = this.isFavorite === 'true';
+    this.id = this.spaceId;
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue]:
```suggestion
```

File: webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue [hunk 6]
```diff
@@ -36,29 +40,26 @@ export default {
     },
   },
   data: () => ({
-    isFavorite: false,
+    favorite: false,
+    id: ''
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue]:
```suggestion
```

File: webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue [hunk 7]
```diff
@@ -1,8 +1,8 @@
 <template>
   <favorite-button
     :id="spaceId"
-    :space-id="spaceId"
-    :favorite="isFavorite"
+    :space-id="id"
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue]:
```suggestion
    :space-id="spaceId"
```

File: webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue [hunk 8]
```diff
@@ -36,29 +40,26 @@ export default {
...
-    this.isFavorite = this.space?.isFavorite === 'true';
+    this.favorite = this.isFavorite === 'true';
+    this.id = this.spaceId;
   },
   methods: {
     removed() {
       this.displayAlert(this.$t('Favorite.tooltip.SuccessfullyDeletedFavorite', {0: this.$t('spaceList.alert.label')}));
       this.$emit('removed');
-      document.dispatchEvent(new CustomEvent('space-favorite-removed', {detail: this.space.id}));
+      document.dispatchEvent(new CustomEvent('space-favorite-removed', {detail: this.id}));
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue]:
```suggestion
      document.dispatchEvent(new CustomEvent('space-favorite-removed', {detail: this.spaceId}));
```

File: webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue [hunk 9]
```diff
@@ -36,29 +40,26 @@ export default {
...
     },
     removeError() {
       this.displayAlert(this.$t('Favorite.tooltip.ErrorDeletingFavorite', {0: this.$t('spaceList.alert.label')}), 'error');
     },
     added() {
       this.displayAlert(this.$t('Favorite.tooltip.SuccessfullyAddedAsFavorite', {0: this.$t('spaceList.alert.label')}));
       this.$emit('added');
-      document.dispatchEvent(new CustomEvent('space-favorite-added', {detail: this.space.id}));
+      document.dispatchEvent(new CustomEvent('space-favorite-added', {detail: this.id}));
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/spaces-list/components/ExoSpaceFavoriteAction.vue]:
```suggestion
      document.dispatchEvent(new CustomEvent('space-favorite-added', {detail: this.spaceId}));
```

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp [hunk 10]
```diff
@@ -23,11 +30,18 @@
...
   int membersNumber= 0;
+  boolean isFavorite= false;
+  boolean isMember =false;
   List<Profile> managers = new ArrayList<>();
   String spaceDescription= "";
   Space space = SpaceUtils.getSpaceByContext();
+
+  FavoriteService favoriteService = ExoContainerContext.getService(FavoriteService.class);
+  String authenticatedUser = ConversationState.getCurrent().getIdentity().getUserId();
+
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp]: To move in **else** block please

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp [hunk 11]
```diff
@@ -23,11 +30,18 @@
...
   PortalRequestContext requestContext = ((PortalRequestContext) RequestContext.getCurrentInstance());
   IdentityManager identityManager = CommonsUtils.getService(IdentityManager.class);
+  Identity userIdentity = identityManager.getOrCreateUserIdentity(authenticatedUser);
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/bannerLogo.jsp]: To move in **else** block please

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**